data/src/main/java/com/example/data/di/WeatherDataModule.kt

```diff
@@ -19,18 +25,49 @@ class WeatherDataModule(private val context: Context) {
 
     @Provides
     @Singleton
-    fun provideFetchWeatherDataUseCase(weatherDataRepository: WeatherDataRepository) =
-        FetchWeatherDataUseCase(weatherDataRepository)
+    fun provideFetchWeatherDataUseCase(currentWeatherRepository: CurrentWeatherRepository) =
```

Comment:
UseCases and mappers shouldn't be a singleton. It is easy and fast to create new instances. And they don't keep anything that we need all the time (like a memory cache). So, we don't need to keep them in memory permanently.

Reply:
Fixed